src/transforms/api/_transform.py

```diff
@@ -218,10 +273,95 @@ def filesystem(self):
             FileSystem:
                 A `FileSystem` object for reading from `Foundry`.
         """
-        return FileSystem(
-            base_path=get_dataset_path(
-                foundry_dev_tools.config.Configuration["cache_dir"],
-                self._dataset_identity,
+        return FileSystem(base_path=self._local_path)
+
+
+class LightweightTransformInput(TransformInput):
+    """LightweightTransformInput class, passed when using @lightweight decorator."""
+
+    def __init__(self, input_arg: Input):
+        super().__init__(input_arg)
+        self.branch = input_arg.branch
```

Comment:
add `self._input_arg.get_local_path_to_dataset()` here.
Without the call a lightweight transformInput does not make sense anyway, so good to call here.